src/backend/core/schemas.py

```diff
@@ -43,6 +45,14 @@ class DocumentSchema(BaseModel):
         str_min_length=1, str_strip_whitespace=True, use_enum_values=True
     )
 
+    @computed_field
+    @property
+    def embedding(self) -> list:
+        """Generate document embedding if hybrid search is enabled"""
+        if check_hybrid_search_enabled():
+            return embed_document(self)
+        return None
+
```

Comment:
I assume it is a way to raise a validation error on document indexation. I would prefer to handle this directly in the view (see my remarks on the `embed_document` part).